control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -118,19 +128,53 @@ func findTagReference(tags []imagev1.TagReference, name string) *imagev1.TagRefe
 	return nil
 }
 
-var CatalogToImage = map[string]string{
-	"certified-operators": "registry.redhat.io/redhat/certified-operator-index:v4.16",
-	"community-operators": "registry.redhat.io/redhat/community-operator-index:v4.16",
-	"redhat-marketplace":  "registry.redhat.io/redhat/redhat-marketplace-index:v4.16",
-	"redhat-operators":    "registry.redhat.io/redhat/redhat-operator-index:v4.16",
+func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullSecret []byte) (map[string]string, error) {
+	imageRef := hcp.Spec.ReleaseImage
+	imageConfig, _, _, err := registryclient.GetMetadata(ctx, imageRef, pullSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get image metadata: %w", err)
+	}
+
+	version := strings.Split(imageConfig.Config.Labels["io.openshift.release"], ".")
+	if len(version) < 2 {
+		return nil, fmt.Errorf("invalid OpenShift release version format: %s", imageConfig.Config.Labels["io.openshift.release"])
+	}
+	major := version[0]
+	minor := version[1]
+	minorInt, err := strconv.Atoi(version[1])
+	if err != nil {
+		return nil, fmt.Errorf("invalid minor version: %w", err)
+	}
+
+	operators := map[string]string{
+		"certified-operators": fmt.Sprintf("registry.redhat.io/redhat/certified-operator-index:v%s.%s", major, minor),
+		"community-operators": fmt.Sprintf("registry.redhat.io/redhat/community-operator-index:v%s.%s", major, minor),
+		"redhat-marketplace":  fmt.Sprintf("registry.redhat.io/redhat/redhat-marketplace-index:v%s.%s", major, minor),
+		"redhat-operators":    fmt.Sprintf("registry.redhat.io/redhat/redhat-operator-index:v%s.%s", major, minor),
+	}
+
+	for i := 0; i < 4; i++ {
+		_, err = registryclient.GetListDigest(ctx, operators["certified-operators"], pullSecret)
+		if err == nil || ctx.Value(TestKey) == "test-reconcile-olm" {
+			break
+		}
```

Comment:
`ctx.Value(TestKey) == "test-reconcile-olm"` this is an unusual way to unit test, why do we need this?

also, please let's move it out of the loop at least for now.

Reply:
yes, this is still unit test aware.  we need to refactor so this is not required.